modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/portlet/FDSSamplePortlet.java

```diff
@@ -117,10 +118,15 @@ private synchronized void _generate(long companyId) throws Exception {
 				Arrays.asList(
 					ObjectFieldUtil.createObjectField(
 						"Text", "String", true, false, null, "Title", "title",
+						ObjectFieldSettingTestUtil.getObjectFieldSettings(
+							"Text"),
```

Comment:
This solution seems to put some burden on the client code using the ObjectFieldUtil class, more specifically:

- Dev has to temember to add the object field settings manually, for all fields being created
- Dev has to make sure the type they pass to the `ObjectFieldSettingTestUtil` is the same as the one being created, this repeating the "Text" string twice: a simple mistake can ruin the field creation
- Dev has to use a TestUtil class in the main Util API

If the settings are mandatory for all fields, it looks more natural that they are added in all API methods right? more specifically, [here](https://github.com/liferay/liferay-portal/blob/4bcbced0ec732aafcc4afe9ea3a865c238e46b1c/modules/apps/object/object-api/src/main/java/com/liferay/object/util/ObjectFieldUtil.java#L37). This way, client code would not need to do the same thing over and over again.

Reply:
Hi @dsanz,

Actually, the `showCounter` setting is mandatory for Text field only, and the client can change its value or even include other allowed settings, for example:

```
"objectFieldSettings": [
    {
        "name": "showCounter",
        "value": "true"
    },
    {
        "name": "maxLength",
        "value": 100
    }
],
```
That's why we couldn't include in the API method

But I understand your point and I will discuss this with the team.

/cc @rodrigopaulino